react-components/src/hooks/useGroundPlaneFromScene.tsx

```diff
@@ -88,14 +89,10 @@ export const useGroundPlaneFromScene = (sceneExternalId: string, sceneSpaceId: s
       geometry.name = `CogniteGroundPlane`;
 
       const mesh = new Mesh(geometry, material);
-      mesh.position.set(
-        groundPlane.translationX,
-        groundPlane.translationY,
-        groundPlane.translationZ
-      );
-      mesh.rotation.set(-Math.PI / 2, 0, 0);
 
-      mesh.position.applyMatrix4(CDF_TO_VIEWER_TRANSFORMATION);
+      const matrix4 = transformation3dToMatrix4(groundPlane).multiply(CDF_TO_VIEWER_TRANSFORMATION);
```

Comment:
Cant we just convert Euler rotation to quaternion and apply to the mesh?

Reply:
We should also apply the scale. Isn't this what I'm doing? I am open to changing the syntax — do you have a concrete suggestion?

Reply:
```
   // Apply position from groundPlane
    mesh.position.set(groundPlane.position.x, groundPlane.position.y, groundPlane.position.z);

    // Convert Euler rotation to quaternion and apply to the mesh
    const euler = new Euler(groundPlane.rotation.x, groundPlane.rotation.y, groundPlane.rotation.z, groundPlane.rotation.order ?? 'XYZ');
    mesh.quaternion.setFromEuler(euler);

    // Apply scaling from groundPlane
    mesh.scale.set(groundPlane.scaleX, groundPlane.scaleY, groundPlane.scaleZ);
```
I feel this is much simpler, unless it causes an issue

Reply:
After chatting with @haakonflatval-cognite, I think maybe the simplest way of doing this is what I updated it to now:

```typescript
      mesh.matrix.copy(matrix4);
      mesh.matrixAutoUpdate = false;
```

Do you agree?